src/Sql/Dirt/dbo/Tables/RiskInsightCriticalApplication.sql

```diff
@@ -0,0 +1,12 @@
+CREATE TABLE [dbo].[RiskInsightCriticalApplication] (
+    [Id]                       UNIQUEIDENTIFIER NOT NULL,
+    [OrganizationId]           UNIQUEIDENTIFIER NOT NULL,
+    [Applications]             NVARCHAR(MAX)    NOT NULL,
+    [CreationDate]             DATETIME2 (7)    NOT NULL,
+    [RevisionDate]             DATETIME2 (7)    NOT NULL,
+    CONSTRAINT [PK_RiskInsightCriticalApplication] PRIMARY KEY CLUSTERED ([Id] ASC),
+    CONSTRAINT [FK_RiskInsightCriticalApplication_Organization] FOREIGN KEY ([OrganizationId]) REFERENCES [dbo].[Organization] ([Id])
```

Comment:
```suggestion
    CONSTRAINT [FK_RiskInsightCriticalApplication_Organization] FOREIGN KEY ([OrganizationId]) REFERENCES [dbo].[Organization] ([Id]) ON DELETE CASCADE
```
This is necessary because, there is an option to delete an `Organization` and, when an org is deleted, it should delete these records as well.

Reply:
Guidance here is for direct children of `Organization` to use our deletion proc for SQL Server e.g. https://github.com/bitwarden/server/blob/main/src/Sql/dbo/Stored%20Procedures/Organization_DeleteById.sql#L87 -- this results in less computational pressure on the DB when an organization is actually deleted. Change that proc and make a new one that's called there that deletes by organization ID.

For EF however, ensure that the entity type configuration does indicate this; we let self-hosters utilize the cascaded delete as their environments will not be under the pressures ours has.

Reply:
Changed the proc to add these tables and changed the entity type configuration to indicate the cascaded delete.